### PR TITLE
USAGOV-1936: Non-published pages API for Contact Center

### PR DESCRIPTION
**Pull request by DaleMHFrey**
USAGOV-1936: Non-published pages API for Contact Center

https://cm-jira.usa.gov/browse/USAGOV-1936

## Description
- Changed the outputted label of "deleted_time" to "updated_time"
- Added in a new route to show nodes that exists in the system, but are set to unpublished and are Basic-Pages

## Type of Changes

- [ ] New Feature
- [ ] Bugfix
- [ ] Frontend (Twig, Sass, JS)
  - Add screenshot showing what it should look like
- [ ] Drupal Config (requires "drush cim")
- [ ] New Modules (requires rebuild)
- [ ] Documentation
- [ ] Infrastructure
  - [ ] CMS
  - [ ] WAF
  - [ ] Egress
  - [ ] Tools
- [x] Other

## Testing Instructions
After merging this in branch, goto the URL path of `/jsonapi/archived-nodes` and you should see a list of Basic-Pages that have a status set to 0, even when in incognito-mode (as an anonymous user).
Also, when going to the URL of`/jsonapi/deleted-nodes` no "deleted_time" should be seen anymore. Instead "updated_time" should be seen in the JSON.

### Change Requirements

- [ ] Requires New Documentation (Link: {})
- [x] Requires New Config - The Contact Center Team is probably going to need to be told that the "deleted_time" field-name is being changed to "updated_time", and they will also need to be made away of this new API-endpoint.
- [ ] Requires New Content

### Validation Steps

- Test instruction 1
- Test instruction 2
- Test instruction 3

## Security Review

- [ ] Adds/updates software (including a library or Drupal module)
- [ ] Communication with external service
- [ ] Changes permissions or workflow
- [ ] Requires SSPP updates

## Reviewer Reminders

- Reviewed code changes
- Reviewed functionality
- Security review complete or not required

## Post PR Approval Instructions

Follow these steps as soon as you merge the new changes.

1. Go to the [USAGov Circle CI project](https://app.circleci.com/pipelines/github/usagov/usagov-2021).
2. Find the commit of this pull request.
3. Build and deploy the changes.
4. Update the Jira ticket by changing the ticket status to `Review in Test` and add a comment. State whether the change is already visible on [cms-dev.usa.gov](http://cms-dev.usa.gov/) and [beta-dev.usa.gov](http://beta-dev.usa.gov/), or if the deployment is still in process.

### CODE REVIEW

File: web/modules/custom/usa_contact_center_api/src/Controller/USAContactCenterController.php [hunk 1]
```diff
@@ -55,6 +55,47 @@ public function getDeletedNodes() {
       $query->condition('deleted_time', $date, '>=');
     }
     $result = $query->execute()->fetchAll();
+
+    // We want to show a property of "updated_time" instead of "deleted_time" as per USAGOV-1936.
+    foreach ( $result as &$item ) {
+      $item = (array)$item;
+      $item['updated_time'] = $item['deleted_time'];
```
Comment (akf) [inline web/modules/custom/usa_contact_center_api/src/Controller/USAContactCenterController.php]: My note was unclear. This should stay as-is (provide 'deleted_time' in the API).

**Review by akf — CHANGES_REQUESTED** [review]
I haven't fully checked this out yet -- just noting that I didn't mean we should change the Deleted API.

**Review by akf — CHANGES_REQUESTED** [review]
Functionality is good. I see that we can also provide an ?archived= timestamp to limit the amount of data returned. 

Two changes are needed: 
1. put the Deleted API back the way it was ("deleted_time", not "updated_time"). This was a misunderstanding. 
2. make it pass the linter. 

Thanks!

File: web/modules/custom/usa_contact_center_api/src/Controller/USAContactCenterController.php (hunk 1 above)
Comment (DaleMHFrey) [reply web/modules/custom/usa_contact_center_api/src/Controller/USAContactCenterController.php]: Ah, I understand now - this is now fixed

**Comment by DaleMHFrey** [comment]
> Functionality is good. I see that we can also provide an ?archived= timestamp to limit the amount of data returned.
> 
> Two changes are needed:
> 
> 1. put the Deleted API back the way it was ("deleted_time", not "updated_time"). This was a misunderstanding.
> 2. make it pass the linter.
> 
> Thanks!

This is now updated & fixed

**Review by akf — APPROVED** [review]

**Pull request merged**